fix: Users removed from the team are displayed as 'Default'

makingthematrix opened the pull request:
fixes: https://github.com/wireapp/android-project/issues/285

This is a temporary solution discussed with Design. In the future we will come up with something else, but now we just want to be consistent with what iOS uses.

Until now, when a user deleted their account, we simply marked her or his data  as 'deleted' in the databases of other users, connected to that one. But if it was a team user, we actually deleted the data in SE, while still expecting in UI to have it.
The SE part of this fix makes removed team users also marked as 'deleted'.
The UI part changes how deleted users' names are displayed: to 'Default'.

SE part: https://github.com/wireapp/wire-android-sync-engine/pull/435

#### APK
[Download build #11826](http://192.168.120.33:8080/job/Pull%20Request%20Builder/11826/artifact/build/artifact/wire-dev-PR1824-11826.apk)
[Download build #11827](http://192.168.120.33:8080/job/Pull%20Request%20Builder/11827/artifact/build/artifact/wire-dev-PR1824-11827.apk)
[Download build #11878](http://192.168.120.33:8080/job/Pull%20Request%20Builder/11878/artifact/build/artifact/wire-dev-PR1824-11878.apk)
[Download build #11882](http://192.168.120.33:8080/job/Pull%20Request%20Builder/11882/artifact/build/artifact/wire-dev-PR1824-11882.apk)
[Download build #11883](http://192.168.120.33:8080/job/Pull%20Request%20Builder/11883/artifact/build/artifact/wire-dev-PR1824-11883.apk)
[Download build #11889](http://192.168.120.33:8080/job/Pull%20Request%20Builder/11889/artifact/build/artifact/wire-dev-PR1824-11889.apk)
[Download build #11988](http://192.168.120.33:8080/job/Pull%20Request%20Builder/11988/artifact/build/artifact/wire-dev-PR1824-11988.apk)
[Download build #12008](http://192.168.120.33:8080/job/Pull%20Request%20Builder/12008/artifact/build/artifact/wire-dev-PR1824-12008.apk)
[Download build #12009](http://192.168.120.33:8080/job/Pull%20Request%20Builder/12009/artifact/build/artifact/wire-dev-PR1824-12009.apk)
[Download build #12088](http://192.168.120.33:8080/job/Pull%20Request%20Builder/12088/artifact/build/artifact/wire-dev-PR1824-12088.apk)
[Download build #12089](http://192.168.120.33:8080/job/Pull%20Request%20Builder/12089/artifact/build/artifact/wire-dev-PR1824-12089.apk)
[Download build #12117](http://192.168.120.33:8080/job/Pull%20Request%20Builder/12117/artifact/build/artifact/wire-dev-PR1824-12117.apk)
[Download build #12171](http://192.168.120.33:8080/job/Pull%20Request%20Builder/12171/artifact/build/artifact/wire-dev-PR1824-12171.apk)
[Download build #12213](http://192.168.120.33:8080/job/Pull%20Request%20Builder/12213/artifact/build/artifact/wire-dev-PR1824-12213.apk)
[Download build #12229](http://192.168.120.33:8080/job/Pull%20Request%20Builder/12229/artifact/build/artifact/wire-dev-PR1824-12229.apk)

red-avtovo commented:
Bug 1:
After member was removed from the team, his avatar and name just disappears

red-avtovo commented:
![screenshot_20181018-111648](https://user-images.githubusercontent.com/10682826/47144322-81095f00-d2c7-11e8-88e6-b8a458f9de1f.png)

red-avtovo commented:
Bug 2
There is no system message "<User> was removed from the team"

red-avtovo commented:
Bug 3
Avatar of deleted user is not reset to default

red-avtovo commented:
![screenshot_20181018-163006](https://user-images.githubusercontent.com/10682826/47162121-add36b80-d2f3-11e8-82ec-a78484c93aab.png)

makingthematrix commented:
I fixed setting the avatar to default, but we don't currently have a system message for "Member was removed from the team" and I think we also lack functionality for it in SE. I would prefer to create a ticket for it and do it in another PR.

red-avtovo commented:
Test

red-avtovo commented on app/src/main/res/values/strings.xml:
```diff
@@ -834,6 +834,9 @@
     <string name="no_conversation_in_list__message">Start a conversation or create a group.</string>
 
+    <string name="user_deleted_username">Default</string>
+    <string name="user_deleted_initial">D</string>
```
to Dev reviewer:
Should we really put hardcoded literal for Default user or we can reuse generic way of initials expression

red-avtovo commented:
Bug 5:
Deleted user is still visible is search results. It allows to try to create a conversation with him and receive an error. Also it is possible to add this person back to conversation and receive even worse error message

red-avtovo commented:
![screenshot_20181030-110956](https://user-images.githubusercontent.com/10682826/47711058-96ba4500-dc34-11e8-9276-fe98e115f01e.jpg)

red-avtovo commented:
I will stop testing this PR unless Devs will finish their review

makingthematrix commented:
I removed deleted users form the search results, changed the initials, so that they're parsed from the default username instead of having their own string. Also when the user goest to a conversation with a deleted user, it is now impossible to access the participant's details.

makingthematrix commented:
This PR was rebased on top of `public_logging`. Please note that for it to work correctly, the SE part of `public_logging` has to include the SE part of `default_username` (already in `develop`).

makingthematrix commented on app/src/main/res/values/strings.xml:
```diff
@@ -1009,6 +1011,8 @@
     <string name="guest_indicator">GUEST</string>
     <string name="participant_tab_guest_indicator_label">Guest</string>
 
+    <string name="conversation_list__def_conv_name">Default</string>
```
done

red-avtovo commented:
-> Bug 1

makingthematrix commented:
build again

makingthematrix commented:
build again

red-avtovo commented:
I can't login in the client

red-avtovo commented:
![screenshot_20190107-174736](https://user-images.githubusercontent.com/10682826/50781074-d2acfe80-12a4-11e9-888d-4cb928943609.png)

Button on the bottom is not clickable

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/wireapp/wire-android-reloaded?pullRequest=1824) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/wireapp/wire-android-reloaded?pullRequest=1824) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/wireapp/wire-android-reloaded?pullRequest=1824) it.</sub>

AndroidBob commented:
Build [1](https://10.10.124.134/job/wire-android-reloaded/job/PR-1824/1/) **failed**.

AndroidBob commented:
Build [2](https://10.10.124.134/job/wire-android-reloaded/job/PR-1824/2/) **failed**.

AndroidBob commented:
Build [3](https://10.10.124.134/job/wire-android-reloaded/job/PR-1824/3/) **failed**.

vitorhugods commented:
I'm closing this PR as there are no clear goals or description, and it's fiddling with our CI scripts.

If there's any contribution in your mind, please be clear when opening the PR.
